_posts/2021-03-12-randomization-unit-analysis-unit.md

```diff
@@ -0,0 +1,212 @@
+
+
+These variance estimations can then be plugged into the same Z-test as above:
+
+$$ Z = \frac{\hat{p}_T-\hat{p}_C}{\sqrt{\frac{Var({p}_T)}{n_T}+\frac{Var({p}_C)}{n_C}}} $$
+
+Where $$ n_C $$ and $$ n_T $$ are the number of users in control and treatment, respectively. 
+
+### Bootstrapping
+
+Bootstrapping is another method we can use to obtain a proper variance estimate and p-value. Bootstrapping is intuitive and easy to understand, but more computationally expensive to calculate.
+
+<p align="center">
+    <img width="100%" src="{{ site.baseurl }}{% link images/randomization-unit-analysis-unit/bootstrapping.png %}">
+</p>
+
+Under the null hypothesis, there is no difference between the control and treatment groups. Therefore, to generate samples under the null hypothesis, we can mix up all our users and then:
+
+1. Randomly create a new control & treatment group through sampling with replacement from the "mixed" population
+2. Calculate the overall session conversion rate for each group
+3. Calculate the difference in conversion rates between the two groups ($$ p_T - p_C $$)
+4. Repeat (1-3) thousands of time and store the results
+
+We now have a bootstrapped distribution of expected differences under the null hypothesis, which we can use in conjunction with the observed difference from the original control & treatment splits to calculate a p-value.
+
+### Results
+
+#### Varying the sessions per user distribution
+
+Applying these two methods to the changing sessions per user distribution, we can see the bootstrap & delta method p-values maintain the expected uniform distributions even as the skew in sessions per user increases:
+
+<p align="center">
+    <img width="100%" src="{{ site.baseurl }}{% link images/randomization-unit-analysis-unit/p_values_session_skew_all.gif %}">
+</p>
+
+#### Varying the conversion rate with sessions per user
+
+We can again see the superior performance of hte bootstrap & delta method p-values while varying the change in conversion rate as a function of sessions per user, as evident by their uniform distributions under all scenarios.
+
+<p align="center">
+    <img width="100%" src="{{ site.baseurl }}{% link images/randomization-unit-analysis-unit/p_values_cvr_decay_factor_all.gif %}">
+</p>
+
+## User grain metrics
+
+As an alternative to changing your statistical test, you can instead change your analysis metric so that it is at the same grain as your randomization unit. With the example from this post, this would involve looking at user conversion rate instead of session conversion rate. User conversion rate would be defined as the proportion of users who converted **at least once**. You may choose to look at conversions per user instead, if you believe your treatment may change how often users convert. Note that this would involve a different statistical test, such as the t-test, as we would no longer be dealing with comparing two binomial proportions.
```

Comment:
```suggestion
As an alternative to changing your statistical test, you can instead change your analysis metric so that it is at the same grain as your randomization unit. With the example from this post, this would involve looking at user conversion rate instead of session conversion rate. User conversion rate would be defined as the proportion of users who converted **at least once**. You may choose to look at conversions per user instead, if you believe your treatment may not only change whether users convert, but how often they do. Note that this would involve a different statistical test, such as the t-test, as we would no longer be comparing two binomial proportions.
```